src/cdef.rs

```diff
@@ -116,9 +116,10 @@ fn constrain(diff: i32, threshold: i32, damping: i32) -> i32 {
 // Unlike the AOM code, our block addressing points to the UL corner
 // of the 2-pixel padding around the block, not the block itself.
 // The destination is unpadded.
-fn cdef_filter_block(dst: &mut [u16], dstride: i32, input: &[u16], istride: i32,
-                     pri_strength: i32, sec_strength: i32, dir: usize, pri_damping: i32,
-                     sec_damping: i32, xsize: i32, ysize: i32, coeff_shift: i32) {
+unsafe fn cdef_filter_block(dst: &mut [u16], dstride: isize, input: &[u16],
```

Comment:
Would be sort of nicer if you mark only the loop unsafe.

Reply:
The motivation for lifting it to the function level was to avoid adding another level of nesting.
This approach minimizes changed lines while admitting optimization of the inner loop.